### PR TITLE
Solana dex.trades (Lifinity v1/v2, Raydium v3, Phoenix v1, orca)

### PR DESCRIPTION
**Pull request by andrewhong5297**
Andddd adding four dexes:
- lifinity v1 + v2
- raydium v3
- phoenix v1

Relies on us getting PR #4476 merged first to deal with decimal accuracy issues. Then we can merge this one.

Also includes dex.solana_trades, with the five dex models we have now (which make up 95% of dex volume on Solana)

### CODE REVIEW

**Comment by andrewhong5297** [comment]
@jeff-dude @antonio-mendes now that the dynamic solana fungible tokens spell was merged, this one is good to go too.

File: models/phoenix/phoenix_v1_trades.sql
Comment (jeff-dude) [inline models/phoenix/phoenix_v1_trades.sql:1]: assuming similar feedback on this one too
- incremental filters where possible
- consider metadata joins here vs. downstream
- version column design / datatype
- any possible hardcoded tables, if any

File: models/raydium/raydium_v3_trades.sql
Comment (jeff-dude) [inline models/raydium/raydium_v3_trades.sql:1]: at first glance, looks like possibly same feedback here too

**Review by jeff-dude — CHANGES_REQUESTED** [review]
left a few comments below on this one too

File: models/lifinity/lifinity_v1_trades.sql
Comment (andrewhong5297) [reply models/lifinity/lifinity_v1_trades.sql:41]: Unlike the NFT ones, these don't change, so it's safe to keep I would say.

File: models/lifinity/lifinity_v1_trades.sql
Comment (andrewhong5297) [reply models/lifinity/lifinity_v1_trades.sql:53]: for nft trades it felt too late lol, I'll use int for dexes since they have a lot more versions

File: models/lifinity/lifinity_v1_trades.sql
Comment (andrewhong5297) [reply models/lifinity/lifinity_v1_trades.sql:46]: can we use incremental macro? does it to the "else" logic on project start date?

File: models/raydium/raydium_v3_trades.sql
Comment (andrewhong5297) [inline models/raydium/raydium_v3_trades.sql:96]: @jeff-dude @aalan3 how do I pass a table alias into this predicate? this does not seem to be working

Comment (aalan3) [reply models/raydium/raydium_v3_trades.sql:96]: `WHERE` -> `AND` you are still inside the join here :)

Comment (andrewhong5297) [reply models/raydium/raydium_v3_trades.sql:96]: ah yeah, it's failing on other models like Phoenix too though

File: models/phoenix/phoenix_v1_trades.sql
Comment (andrewhong5297) [inline models/phoenix/phoenix_v1_trades.sql:103]: @aalan3 this one

Comment (aalan3) [reply models/phoenix/phoenix_v1_trades.sql:103]: try using `l. call_block_time` here? but this should be working.

Comment (aalan3) [reply models/phoenix/phoenix_v1_trades.sql:103]: the line numbers you see in the failure doesn't map very well to the uncompiled sql

Comment (andrewhong5297) [reply models/phoenix/phoenix_v1_trades.sql:103]: okay, trying CI again

Comment (andrewhong5297) [reply models/phoenix/phoenix_v1_trades.sql]: alright it worked this time!

**Comment by andrewhong5297** [comment]
@jeff-dude this is good to go now

**Comment by andrewhong5297** [comment]
wait don't merge this, I'm seeing some weird values in orca whirlpools spell

**Comment by jeff-dude** [comment]
> wait don't merge this, I'm seeing some weird values in orca whirlpools spell

i will save final review on both PRs for tomorrow 👍 won't be able to merge rn anyway, with other items going through

**Comment by andrewhong5297** [comment]
fyi orca table will meed to be rerun

**Comment by jeff-dude** [comment]
> fyi orca table will meed to be rerun

anything modified will do full refresh

File: models/raydium/raydium_v3_trades.sql
Comment (jeff-dude) [inline models/raydium/raydium_v3_trades.sql:110]: in refactor, ensure we update to jinja syntax for lineage

**Review by jeff-dude — APPROVED** [review]
approving with comments below for follow-up refactor

**Pull request merged**
